ingest/src/commands/pages.ts

```diff
@@ -16,68 +21,80 @@ const commandModule: CommandModule<
+      string: true,
+      description: "A source name to load. If unspecified, loads all sources.",
+    });
   },
   handler: async ({ source }) => {
-    const requestedSources = new Set(Array.isArray(source) ? source : [source]);
-
-    const {
-      DEVCENTER_CONNECTION_URI,
-      MONGODB_CONNECTION_URI,
-      MONGODB_DATABASE_NAME,
-    } = assertEnvVars(INGEST_ENV_VARS);
-
-    const snootySources = await Promise.all(
-      snootyProjects.map(({ project, baseUrl }) =>
-        makeSnootyDataSource({
-          baseUrl,
-          manifestUrl: `https://snooty-data-api.mongodb.com/projects/${project}/master/documents`,
-          name: `snooty-${project}`,
-        })
-      )
-    );
-
-    const devCenterSource = await makeDevCenterDataSource({
-      name: "devcenter",
-      collectionName: "search_content_prod",
-      databaseName: "devcenter",
-      connectionUri: DEVCENTER_CONNECTION_URI,
-      baseUrl: "https://www.mongodb.com/developer",
-    });
+    const { MONGODB_CONNECTION_URI, MONGODB_DATABASE_NAME } =
+      assertEnvVars(INGEST_ENV_VARS);
 
-    const availableSources = [...snootySources, devCenterSource];
-
-    const sources = availableSources.filter(({ name }) =>
-      requestedSources.has(name)
-    );
-
-    const pageStore = await makeDatabaseConnection({
+    const store = await makeDatabaseConnection({
       connectionUri: MONGODB_CONNECTION_URI,
       databaseName: MONGODB_DATABASE_NAME,
     });
 
     try {
-      if (sources.length === 0) {
-        throw new Error(
-          `Request at least one valid source. Available sources:\n${availableSources
-            .map(({ name }) => `- ${name}`)
-            .join("\n")}`
-        );
-      }
-
-      logger.info(
-        `Loaded sources:\n${sources.map(({ name }) => `- ${name}`).join("\n")}`
-      );
-
-      await updatePages({
-        sources,
-        pageStore,
-      });
+      await doPagesCommand({ source, store });
     } finally {
-      await pageStore.close();
+      await store.close();
     }
   },
   describe: "Update pages data from sources",
 };
 
 export default commandModule;
+
+export const doPagesCommand = async ({
+  source,
+  store,
+}: PagesCommandArgs & {
+  store: PageStore;
+}) => {
+  const { DEVCENTER_CONNECTION_URI } = assertEnvVars(INGEST_ENV_VARS);
+
+  const requestedSources = new Set(Array.isArray(source) ? source : [source]);
+
+  const snootySources = await Promise.all(
+    snootyProjects.map(({ project, baseUrl }) =>
+      makeSnootyDataSource({
+        baseUrl,
+        manifestUrl: `https://snooty-data-api.mongodb.com/projects/${project}/master/documents`,
+        name: `snooty-${project}`,
+      })
+    )
+  );
+
+  const devCenterSource = await makeDevCenterDataSource({
+    name: "devcenter",
+    collectionName: "search_content_prod",
+    databaseName: "devcenter",
+    connectionUri: DEVCENTER_CONNECTION_URI,
+    baseUrl: "https://www.mongodb.com/developer",
+  });
```

Comment:
thought for future work: would be nice to have these pluggable in some way so they're not defined inside of this function..obvi no action here now

Reply:
That is the eventual plan